src/main/kotlin/com/cultureamp/eventsourcing/BatchedAsyncEventProcessor.kt

```diff
@@ -66,6 +67,7 @@ class BatchedAsyncEventProcessor<M : EventMetadata>(
     )
 
     fun processOneBatch(): Action {
+        batchLockAcquisition(bookmarkName)
```

Comment:
I wonder if it might be worth naming this something like "keepAlive" or "ensureConnectionIsAlive" to disambiguate where the true locking happens?

Reply:
I think that is an implementation detail - it's not the event processor that is determining how the locking works. But perhaps I should call it `acquireLock` so it's not misleadingly stating that it will be a lock per batch. WDYT?

Reply:
Hmm yeah, anything to just make it clear that it's not a lock per batch